FlowCrypt/Controllers/Setup/SetupInitialViewController.swift

```diff
@@ -131,6 +141,38 @@ extension SetupInitialViewController {
         handleCommon(error: error)
         state = .error(error)
     }
+
+    private func checkClientConfigurationIntegrity() {
+        guard let integrityErrorMessage = clientConfigurationService.checkForUsingKeyManager() else {
```

Comment:
Here if error is `nil`, which means that user is not using a key manager, we start to check EKM keys. Is that right?

Reply:
You are commingling error checking application logic here, it seems wrong. I didn't notice it earlier but this comment:

```
/// - Returns: Error message if not using key manager, returns nil if using key manager
```

Shows a badly designed method signature for this usecase. It suggests there are only two states: either there is error, or EKM must be used. This is false. There are in fact three states:

- EKM is in use because `organisationalRules.isUsingKeyManager` and other OrgRules are consistent with it (result: no error, use EKM)
- EKM is in use because `organisationalRules.isUsingKeyManager` and other OrgRules are NOT consistent with it (result: error)
- EKM is not in use because `organisationalRules.isUsingKeyManager`

Not only that, if we call something `checkClientConfigurationIntegrity` should just be doing what it says - checking integrity. But here, it's first checking integrity, then deciding which login flow to use, then changing state..

We need to call things by exactly what they do.

Ask more questions if unsure.